server/master/src/main/java/org/apache/accumulo/master/FateServiceHandler.java

```diff
@@ -697,10 +700,29 @@ private TableId validateTableIdArgument(ByteBuffer tableIdArg, TableOperation op
     }
   }
 
-  // Verify table name arguments are valid, and match any additional restrictions
+  // Verify older table name arguments are valid, and match any additional restrictions
   private String validateTableNameArgument(ByteBuffer tableNameArg, TableOperation op,
       Validator<String> userValidator) throws ThriftTableOperationException {
     String tableName = tableNameArg == null ? null : ByteBufferUtil.toString(tableNameArg);
+    if ((tableName != null) && (tableName.length() > MAX_TABLE_NAME_LEN)) {
+      log.warn("Table names greater than " + MAX_TABLE_NAME_LEN
+          + " characters should be renamed to conform to a " + MAX_TABLE_NAME_LEN
+          + " character limit. Longer table names are no longer supported and may result in "
+          + " unexpected behavior.");
+    }
```

Comment:
As follow-on, it might be useful to create a cache of recent warnings, so tables which aren't immediately renamed, but are relatively active, don't spam the logs too much.